test/e2e/util/util.go

```diff
@@ -983,3 +986,43 @@ func RunQueryAtTime(ctx context.Context, log logr.Logger, prometheusClient prome
 		},
 	}, nil
 }
+
+func EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations(t *testing.T, ctx context.Context, hostClient crclient.Client, hcpNs string) {
+	t.Run("EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations", func(t *testing.T) {
+		g := NewWithT(t)
+		hcpPods := &corev1.PodList{}
+		if err := hostClient.List(ctx, hcpPods, &client.ListOptions{
+			Namespace: hcpNs,
+		}); err != nil {
+			t.Fatalf("cannot list hostedControlPlane pods: %v", err)
+		}
+
+		// Check if the pod's volumes are hostPath or emptyDir, if so, the deployment
+		// should annotate the pods with the safe-to-evict-local-volumes, with all the volumes
+		// involved as an string  and separated by comma, to satisfy CA operator contract.
+		// more info here: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-types-of-pods-can-prevent-ca-from-removing-a-node
+		for _, pod := range hcpPods.Items {
+			localVolumeList := make([]string, 0)
+			if pod.Spec.Volumes != nil {
+				for _, volume := range pod.Spec.Volumes {
+					if volume.EmptyDir != nil {
+						checked := false
+						// Checking if PodSafeToEvictLocalVolumesKey annotation is set
+						if val, exists := pod.Annotations[suppconfig.PodSafeToEvictLocalVolumesKey]; exists {
```

Comment:
this shouldn't be gating on exists to run the validation, 
`if volume.EmptyDir != nil {`
then add name to slice
By the end of the loop validate the annotation is set with the list of names as value